routes/category.js

```diff
@@ -1,26 +1,25 @@
 const express = require('express');
 const router = express.Router();
-const products = require('../src/data/products');
-const categories = require('../src/data/categories');
+const dataService = require('../services/dataService');
 
+/*
+ * This method first gets the category ID from the request params.
+ * Then it loops through the combinedProductMap.
+ * It checks if the category id of that particular product is equal to the request param
+ * If matches then map it in the response
+ */
 router.get('/:ctyId', function(req, res, next) {
-    const ctyId = req.params.ctyId;
-    let currentCategory;
-    let response = [];
-    categories.categories.forEach( category => {
-        if (category.id === ctyId) {
-            currentCategory = category;
-        }
-    });
-    products.products.forEach( product => {
-        if (product.categoryId === ctyId) {
-            response.push({
-                ...product,
-                category: currentCategory
-            });
-        }
-    });
-    res.send(response);
+  const { ctyId } = req.params;
+  const response = {};
+  const combinedProductsMap = dataService.getCombinedProductMap();
+  for (let product in combinedProductsMap) {
+    if (combinedProductsMap.hasOwnProperty(product)) {
```

Comment:
`product` was fetched from the `combinedProductsMap` in the above line, Do we need to check it again here?

Reply:
Yes, my IDE gave me that warning. The best practice is to run `hasOwnProperty()` check to see if the property exists. Here it's kinda implicit that the object has that property.